### ISSUE TITLE
Merge workflows triggered by the same event

### ISSUE THREAD

**Issue by cjolowicz** [cjolowicz/cookiecutter-hypermodern-python#144]
It would be logical to merge workflows which are triggered by the same event, into a single workflow. If the workflows are converted to separate jobs, they can still run in parallel. On the other hand, they could also be merged into a single job, for example if they share many setup steps.

**Comment by cjolowicz** [comment]
See #241

**Comment by cjolowicz** [comment]
This would now apply to two remaining workflows which could be merged into the Tests workflow:

- The Docs workflow
- The Coverage workflow

Maybe this can be done by splitting the Tests workflow into jobs for each Nox session.

The Tests workflow currently runs the following sessions in CI:

- pre-commit
- safety
- mypy
- tests

These don't need to run on all platforms and Python versions:

- Only tests needs to run on Windows and macOS.
- Only tests and mypy need to run on older Python versions.

We would then add the docs session to the matrix, only for Linux with the current Python release.

The coverage upload step can be appended to the workflow, with a condition that the job is running the tests session on Linux with the current Python release.

**Comment by cjolowicz** [comment]
Related PRs:

- #301
